File: movie-frontend/src/app/movie/models/create-movie-request.ts
export class CreateMovieRequest {
  constructor(
    public readonly title: string,
    public readonly description: string,
    public readonly lengthInMinutes: number,
    public readonly releaseYear: number,
  ) {}

  public static fromForm(
    form: CreateMovieRequestRawFormValue,
  ): CreateMovieRequest {
    return new CreateMovieRequest(
      form.title,
      form.description,
      form.lengthInMinutes,
      form.releaseYear,
    );
  }
}

export interface CreateMovieRequestRawFormValue {
  title: string;
  description: string;
  lengthInMinutes: number;
  releaseYear: number;
}

File: movie-backend/src/main/java/com/example/movie/dto/CreateMovieRequest.java
package com.example.movie.dto;

import jakarta.validation.constraints.*;
import lombok.Builder;

@Builder
public record CreateMovieRequest(
        @NotEmpty @Size(min = 3, max = 30) String title,
        @NotEmpty @Size(min = 3, max = 500) String description,
        @NotNull @Min(10) @Max(300) Integer lengthInMinutes,
        @NotNull @Min(1850) @Max(2030) Integer releaseYear
) {
}
